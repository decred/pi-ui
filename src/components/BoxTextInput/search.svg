<svg xmlns="http://www.w3.org/2000/svg" width="15" height="15.537" viewBox="0 0 15 15.537"><defs><style>.a{fill:#8997a5;}</style></defs><path class="a" d="M15.756,14.151l-3.7-3.846a6.27,6.27,0,1,0-4.8,2.24,6.206,6.206,0,0,0,3.594-1.135l3.726,3.875a.818.818,0,1,0,1.179-1.134ZM7.257,1.636A4.637,4.637,0,1,1,2.62,6.273,4.642,4.642,0,0,1,7.257,1.636Z" transform="translate(-0.984)"/></svg>