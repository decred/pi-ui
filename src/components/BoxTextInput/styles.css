.boxtextinput {
  background-color: transparent;
  font-family: var(--font-family-text);
  border: 1px solid var(--input-border-color);
  outline: 0;
  width: 100%;
  font-size: var(--font-size-normal);
  font-weight: var(--font-weight-regular);
  transition: all 0.25s;
  line-height: var(--spacing-2);
  padding: 1rem 2rem;
  padding-right: 4rem;
  margin-top: 1.2rem;
}

.boxTextInputRounded {
  border-radius: 0.5rem;
}

.boxTextInputError {
  border-color: var(--color-orange);
}

.errorMsg {
  visibility: hidden;
  color: var(--color-orange);
  font-size: var(--font-size-small);
  min-height: 2rem;
  margin-top: 0.3rem;
}

.errorMsgActive {
  visibility: visible;
}

.boxtextinput::placeholder {
  color: var(--text-input-color);
}

.boxtextinputWrapper {
  font-size: var(--font-size-normal);
  position: relative;
}

.boxtextinputWrapper input {
  color: var(--text-color);
}

.boxtextinputButton {
  cursor: pointer;
  border: 0;
  outline: 0;
  position: absolute;
  font-size: inherit;
  background: none !important;
  right: 1rem;
  top: calc(50% - 0.5rem);
  transform: translate(-50%, -50%);
  user-select: none;
}
