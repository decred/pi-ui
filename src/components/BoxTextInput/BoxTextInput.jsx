import React from "react";
import PropTypes from "prop-types";
import styles from "./styles.css";
import { classNames } from "../../utils";
import Icon from "../Icon/Icon.jsx";

const FormWrapper = ({ className, onSubmit, children }) => {
  return (
    <form className={className} onSubmit={onSubmit}>
      {children}
    </form>
  );
};

const DefaultWrapper = ({ className, onSubmit, children }) => {
  return (
    <div className={className} onSubmit={onSubmit}>
      {children}
    </div>
  );
};

const BoxTextInput = ({
  value,
  onChange,
  placeholder,
  id,
  onSubmit,
  error,
  rounded,
  searchInput,
  className,
  inputClassName,
  ...props
}) => {
  const handleSubmit = (e) => {
    e && e.preventDefault();
    onSubmit && onSubmit();
  };
  const Wrapper = searchInput ? FormWrapper : DefaultWrapper;
  return (
    <Wrapper
      className={classNames(styles.boxtextinputWrapper, className)}
      onSubmit={handleSubmit}>
      <input
        className={classNames(
          styles.boxtextinput,
          error && styles.boxTextInputError,
          rounded && styles.boxTextInputRounded,
          inputClassName
        )}
        type="text"
        onChange={onChange}
        value={value}
        id={id}
        placeholder={placeholder}
        {...props}
      />
      {searchInput && (
        <button
          type="submit"
          className={styles.boxtextinputButton}
          data-testid="submit-button">
          <Icon type="search" />
        </button>
      )}
      <p
        className={classNames(styles.errorMsg, error && styles.errorMsgActive)}>
        {error}
      </p>
    </Wrapper>
  );
};

FormWrapper.propTypes = {
  className: PropTypes.string,
  onSubmit: PropTypes.func.isRequired,
  children: PropTypes.node,
};

DefaultWrapper.propTypes = {
  className: PropTypes.string,
  onSubmit: PropTypes.func.isRequired,
  children: PropTypes.node,
};

BoxTextInput.propTypes = {
  placeholder: PropTypes.node,
  id: PropTypes.string,
  onSubmit: PropTypes.func,
  error: PropTypes.node,
  rounded: PropTypes.bool,
  searchInput: PropTypes.bool,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  onChange: PropTypes.func,
  className: PropTypes.string,
  inputClassName: PropTypes.string,
};

BoxTextInput.defaultProps = {
  rounded: false,
  searchInput: false,
};

export default BoxTextInput;
