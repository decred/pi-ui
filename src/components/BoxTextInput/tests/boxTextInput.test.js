import React from "react";
import BoxTextInput from "../BoxTextInput";
import { create } from "react-test-renderer";
import {
  defaultLightTheme,
  ThemeProvider,
  DEFAULT_LIGHT_THEME_NAME,
} from "../../../theme";
import { fireEvent, render } from "@testing-library/react";

describe("BoxTextInput component", () => {
  test("Matches the snapshot", () => {
    const boxTextInputForm = create(
      <ThemeProvider
        themes={{ [DEFAULT_LIGHT_THEME_NAME]: defaultLightTheme }}
        defaultThemeName={DEFAULT_LIGHT_THEME_NAME}
      >
        <BoxTextInput searchInput={true} />
      </ThemeProvider>
    );
    expect(boxTextInputForm.toJSON()).toMatchSnapshot();

    const boxTextInputDefault = create(
      <ThemeProvider
        themes={{ [DEFAULT_LIGHT_THEME_NAME]: defaultLightTheme }}
        defaultThemeName={DEFAULT_LIGHT_THEME_NAME}
      >
        <BoxTextInput />
      </ThemeProvider>
    );
    expect(boxTextInputDefault.toJSON()).toMatchSnapshot();
  });

  test("Calls onSubmit() prop on submit button click", () => {
    const mockedOnSubmit = jest.fn();

    const { getByTestId, queryByTestId } = render(
      <ThemeProvider
        themes={{ [DEFAULT_LIGHT_THEME_NAME]: defaultLightTheme }}
        defaultThemeName={DEFAULT_LIGHT_THEME_NAME}
      >
        <BoxTextInput onSubmit={mockedOnSubmit} searchInput={true} />
      </ThemeProvider>
    );

    // submit button rendered
    expect(queryByTestId("submit-button")).toBeTruthy();

    // click submit button
    fireEvent.click(getByTestId("submit-button"));

    // expect onSubmit to be called
    expect(mockedOnSubmit).toHaveBeenCalledTimes(1);
  });
});
