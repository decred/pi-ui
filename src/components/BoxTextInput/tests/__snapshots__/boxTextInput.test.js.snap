// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoxTextInput component Matches the snapshot 1`] = `
<form
  className="boxtextinputWrapper"
  onSubmit={[Function]}
>
  <input
    className="boxtextinput"
    type="text"
  />
  <button
    className="boxtextinputButton"
    data-testid="submit-button"
    type="submit"
  >
    <svg
      height="16"
      viewBox="0 0 450 450"
      width="16"
    >
      <g>
        <g
          fill="#c4cbd2"
        >
          <path
            d="M417.494118,397.996875 L319.552941,289.828125 C369.576866,226.478443 371.45145,134.704401 324.06138,69.1084373 C276.67131,3.5124731 192.746493,-18.2850446 122.219344,16.6846668 C51.6921948,51.6543782 13.5076963,133.997765 30.384328,214.723309 C47.2609597,295.448853 114.665326,352.872082 192.494118,352.828125 C226.553641,352.923437 259.798972,341.768237 287.629412,320.90625 L386.258824,429.890625 C391.59298,435.946438 399.582119,438.482788 407.168761,436.529002 C414.755404,434.575216 420.76494,428.43379 422.897495,420.455077 C425.030049,412.476363 422.955447,403.895673 417.467647,397.996875 L417.494118,397.996875 Z M192.520588,46.0125 C260.310293,46.0125 315.264706,104.401564 315.264706,176.428125 C315.264706,248.454686 260.310293,306.84375 192.520588,306.84375 C124.730884,306.84375 69.7764706,248.454686 69.7764706,176.428125 C69.8494166,104.433692 124.761122,46.0900051 192.520588,46.0125 Z"
          />
        </g>
      </g>
    </svg>
  </button>
  <p
    className="errorMsg"
  />
</form>
`;

exports[`BoxTextInput component Matches the snapshot 2`] = `
<div
  className="boxtextinputWrapper"
  onSubmit={[Function]}
>
  <input
    className="boxtextinput"
    type="text"
  />
  <p
    className="errorMsg"
  />
</div>
`;
